lib/screens/surveydetails/widget/question_paging_widget.dart

```diff
@@ -28,80 +30,95 @@ class _QuestionPagingWidgetState extends State<QuestionPagingWidget> {
+    }
     final PageController pageController = PageController();
-    return Stack(
-      children: [
-        PageView(
-          controller: pageController,
-          onPageChanged: _onPageChanged,
-          physics: const NeverScrollableScrollPhysics(),
-          children: widget.questions
-              .map((QuestionUiModel question) => BackgroundWidget(
-                    image: Image.network(question.largeCoverImageUrl).image,
-                  ))
-              .toList(),
-        ),
-        _buildQuestionContent(
-            context, widget.questions[_currentIndex], pageController)
-      ],
-    );
-  }
 
-  Widget _buildQuestionContent(
-    BuildContext context,
-    QuestionUiModel question,
-    PageController controller,
-  ) {
     return Stack(
       children: [
+        BackgroundWidget(
+          image: Image.network(widget.surveyDetailUiModel.largeCoverImageUrl)
+              .image,
+        ),
         SafeArea(
           child: Padding(
             padding: const EdgeInsets.all(20),
             child: Column(
               crossAxisAlignment: CrossAxisAlignment.start,
               children: [
-                _buildQuestionToolbar(context, question),
-                const SizedBox(height: Dimensions.paddingLarge),
-                _buildQuestionHeader(context, question),
+                _buildQuestionToolbar(
+                  context,
+                  widget.surveyDetailUiModel.questions[_currentIndex],
+                ),
+                _buildPagedQuestions(context, pageController),
                 const Spacer(),
-                _buildQuestionFooter(context, question, controller),
+                _buildQuestionFooter(
+                  context,
+                  widget.surveyDetailUiModel.questions[_currentIndex],
+                  pageController,
+                ),
               ],
             ),
           ),
-        )
+        ),
       ],
     );
   }
 
+  Widget _buildPagedQuestions(
+    BuildContext context,
+    PageController pageController,
+  ) {
+    return Expanded(
+      child: PageView(
+        controller: pageController,
+        onPageChanged: _onPageChanged,
+        physics: const NeverScrollableScrollPhysics(),
+        children: widget.surveyDetailUiModel.questions
+            .map((QuestionUiModel question) =>
+                _buildQuestionHeader(context, question))
+            .toList(),
+      ),
+    );
+  }
+
   Widget _buildQuestionToolbar(BuildContext context, QuestionUiModel question) {
-    return Row(
-      children: [
-        _isIntroQuestion(question)
-            ? MaterialButton(
-                onPressed: () => {context.pop()},
-                minWidth: 30,
-                height: 30,
+    return Padding(
+      padding: const EdgeInsets.only(bottom: Dimensions.paddingMedium),
+      child: SizedBox(
+        height: 44.0,
+        child: Row(
+          children: [
+            _isIntroQuestion(question)
```

Comment:
We can create a local variable to this flag, so we can re-use it, what do you think? 🤔